### DISCUSSION TITLE
DAI 2024-2025 - Java IOs benchmarking - Florian Chollet

### DISCUSSION THREAD

**Discussion by luma2010** [heig-vd-dai-course/heig-vd-dai-course#361] [Show and tell]
https://github.com/luma2010/heig-vd-dai-course-java-io

Result for 1B :
| Implementation | Time (ms) |
| -------- | ------- |
| BINARY WRITE | 99 |
| BINARY READ | 83 |
| BUFFERED_BINARY WRITE    | 82 |
| BUFFERED_BINARY READ    | 81 |
| TEXT WRITE    | 84 |
| TEXT READ    | 100 |
| BUFFERED_TEXT  WRITE    | 77 |
| BUFFERED_TEXT  READ    | 78 |

Result for 1KB :
| Implementation | Time (ms) |
| -------- | ------- |
| BINARY WRITE | 81 |
| BINARY READ | 74 |
| BUFFERED_BINARY WRITE    | 79 |
| BUFFERED_BINARY READ    | 78 |
| TEXT WRITE    | 87 |
| TEXT READ    | 80 |
| BUFFERED_TEXT  WRITE    | 79 |
| BUFFERED_TEXT  READ    | 77 |

Result for 1MiB :
| Implementation | Time (ms) |
| -------- | ------- |
| BINARY WRITE | 669 |
| BINARY READ | 248 |
| BUFFERED_BINARY WRITE    | 96 |
| BUFFERED_BINARY READ    | 96 |
| TEXT WRITE    | 148 |
| TEXT READ    | 135 |
| BUFFERED_TEXT  WRITE    | 111 |
| BUFFERED_TEXT  READ    | 105 |

Result for 5MiB :
| Implementation | Time (ms) |
| -------- | ------- |
| BINARY WRITE | 3158 |
| BINARY READ | 949 |
| BUFFERED_BINARY WRITE    | 142 |
| BUFFERED_BINARY READ    | 142 |
| TEXT WRITE    | 253 |
| TEXT READ    | 255 |
| BUFFERED_TEXT  WRITE    | 176 |
| BUFFERED_TEXT  READ    | 161 |

Conclusion :
The most efficient stream is the buffered binary stream for big file, but for small file it is the buffered text stream.
It is more efficient than the text stream because it doesn't need to encode for big file.
Binary file and text file contain the same things, the difference is that text file will be interpreted.
A character encoding is used to encode 0 and 1 into character like "A" or "?"
The benchmark methodology is important because it shows us the importance of buffer (especially for big file)

**Comment by ludelafo** [comment]
Thank you!

You can always find the solution in the solution repository if you ever need to compare your work with the official one here: https://github.com/heig-vd-dai-course/heig-vd-dai-course-solutions.

Do not hesitate to ask questions if something is not clear, either in this discussion or in the Q&A category of the GitHub Discussions here: https://github.com/orgs/heig-vd-dai-course/discussions/categories/q-a.
